util/chunk/chunk.go

```diff
@@ -320,6 +320,9 @@ func reCalcCapacity(c *Chunk, maxChunkSize int) int {
 	if c.NumRows() < c.capacity {
 		return c.capacity
 	}
+	if c.capacity == 0 {
+		return InitialCapacity
```

Comment:
IMO, this is better that it does not bypass the max chunk size check:

```
if c.capacity == 0 {
		c.capacity = InitialCapacity
} else {
        c.capacity *= 2
}
return mathutil.Min(c.capacity, maxChunkSize)
```

Reply:
Done

Reply:
Well, I find this function `reCalcCapacity` is side-effect free, i.e.  it just calculate the new capacity, without change `c.capacity`

Reply:
@xhebox

Reply:
OK, what about a new variable, or just keep it as-is. I am fine with both.